### PR TITLE
length of topics are different

### PR DESCRIPTION
**Pull request by youngday**
Error will occurred if topic have different length ,so compare length first.

### CODE REVIEW

File: src/pub.rs [hunk 1]
```diff
@@ -165,6 +165,7 @@ impl SocketSend for PubSocket {
         let mut dead_peers = Vec::new();
         for mut subscriber in self.backend.subscribers.iter_mut() {
             for sub_filter in &subscriber.subscriptions {
+                if sub_filter.len() == message.get(0).unwrap().len(){
```
Comment (Alexei-Kornienko) [inline src/pub.rs:168]: @youngday thanks for the pull request. I agree that this is an issue however IMHO your fix is not quite compatible to ZMQ spec:
"SHALL perform a binary comparison of the subscription against the start of the first frame of the message."
So the proper fix would be to 
sub_filter.len() <= ...

**Comment by Alexei-Kornienko** [comment]
Sorry for slow reply. I'm quite busy lately...

**Review by youngday — COMMENTED** [review]
changed  with your advise

**Comment by Alexei-Kornienko** [comment]
Minor formatting issues. Otherwise LGTM. Good enough to merge

**Review by Alexei-Kornienko — APPROVED** [review]

**Pull request merged**